### ISSUE TITLE
type satisfies parameter constraints but throws TypeError

### ISSUE THREAD

**Issue by jrevels** [JuliaLang/julia#30354]
```julia
julia> versioninfo()
Julia Version 1.0.2
Commit d789231e99 (2018-11-08 20:11 UTC)
Platform Info:
  OS: macOS (x86_64-apple-darwin17.5.0)
  CPU: Intel(R) Core(TM) i7-7920HQ CPU @ 3.10GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-6.0.0 (ORCJIT, skylake)

julia> struct Foo{A<:Number,B<:Union{Nothing,Vector{A}}} end

julia> const Bar{B,A} = Foo{A,B}
Foo{A,B} where A where B

julia> Bar{Vector{Int},Int} # this should probably be made to work
ERROR: TypeError: in Foo, in B, expected B<:Union{Nothing, Array{A<:Number,1}}, got Type{Array{Int64,1}}
Stacktrace:
 [1] top-level scope at none:0
```

ref #29848, #24179, #9441, #6383

**Comment by JeffBezanson** [comment]
Seems fixed on master, I believe by #29986.

**Comment by jrevels** [comment]
Ah, yup! Awesome, thanks!
